packages/grafana-data/src/transformations/transformers/filter.ts

```diff
@@ -72,30 +72,28 @@ export const filterFramesTransformer: DataTransformerInfo<FilterOptions> = {
    * Return a modified copy of the series.  If the transform is not or should not
    * be applied, just return the input series
    */
-  transformer: (options: FilterOptions) => {
+  transformer: (options, data) => {
     if (!options.include && !options.exclude) {
-      return noopTransformer.transformer({});
+      return noopTransformer.transformer({}, data);
     }
 
     const include = options.include ? getFrameMatchers(options.include) : null;
     const exclude = options.exclude ? getFrameMatchers(options.exclude) : null;
 
-    return (data: DataFrame[]) => {
-      const processed: DataFrame[] = [];
-      for (const series of data) {
-        if (exclude) {
-          if (exclude(series)) {
-            continue;
-          }
-          if (!include) {
-            processed.push(series);
-          }
+    const processed: DataFrame[] = [];
+    for (const series of data) {
```

Comment:
maybe this can be simplified?

```ts
for (const series of data) {
  if (exclude?.(series)) {
      continue;
  }

  if ((exclude && !include) || include?.(series)) {
    processed.push(series);
  }
}
```

Reply:
Just a quick feedback, you can use the suggestion button (![image](https://user-images.githubusercontent.com/562238/93746849-14571780-fbf6-11ea-92f3-05114dd93ee2.png)) to make suggestions to code and then get credit as a co-author :)

Reply:
Agree with @hugohaggmark , thinks this suggestion makes the code harder to reason about